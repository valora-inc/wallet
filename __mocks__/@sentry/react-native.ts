const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
  startTransaction: jest.fn(),
  nativeCrash: jest.fn(),
  wrap: jest.fn().mockImplementation((x) => x),
}

module.exports = Sentry
