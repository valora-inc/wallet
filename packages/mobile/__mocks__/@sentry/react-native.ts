const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => ({})),
  Severity: {
    Error: 'error',
  },
}

module.exports = Sentry
