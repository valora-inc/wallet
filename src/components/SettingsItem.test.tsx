import { fireEvent, render } from '@testing-library/react-native'
import * as React from 'react'
import 'react-native'
import {
  SettingsItemCta,
  SettingsItemInput,
  SettingsItemSwitch,
  SettingsItemTextValue,
} from 'src/components/SettingsItem'

const title = 'title'
const testID = 'testID'

describe('SettingsItemTextValue', () => {
  const value = 'value'
  it('renders correctly without value', () => {
    const wrapper = render(<SettingsItemTextValue title={title} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const wrapper = render(<SettingsItemTextValue testID={testID} title={title} value={value} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('reacts on press', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <SettingsItemTextValue testID={testID} title={title} value={value} onPress={onPress} />
    )
    fireEvent.press(getByTestId(testID))
    expect(onPress).toHaveBeenCalled()
  })
})

describe('SettingsItemSwitch', () => {
  const value = true
  const onValueChange = jest.fn()
  it('renders correctly', () => {
    const wrapper = render(
      <SettingsItemSwitch title={title} value={value} onValueChange={onValueChange} />
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('reacts on press', () => {
    const { getByTestId } = render(
      <SettingsItemSwitch
        testID={testID}
        title={title}
        value={value}
        onValueChange={onValueChange}
      />
    )
    fireEvent(getByTestId(testID), 'valueChange', !value)
    expect(onValueChange).toHaveBeenCalledWith(!value)
  })
})

describe('SettingsItemInput', () => {
  const value = 'value'
  const newValue = 'newValue'
  const onValueChange = jest.fn()
  it('renders correctly', () => {
    const wrapper = render(
      <SettingsItemInput
        testID={testID}
        onValueChange={onValueChange}
        title={title}
        value={value}
      />
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('reacts on press', () => {
    const { getByTestId } = render(
      <SettingsItemInput
        testID={testID}
        title={title}
        value={value}
        onValueChange={onValueChange}
      />
    )
    fireEvent(getByTestId(testID), 'changeText', newValue)
    expect(onValueChange).toHaveBeenCalledWith(newValue)
  })
})

describe('SettingsItemCta', () => {
  it('renders correctly', () => {
    const { getByTestId, getByText, queryByTestId } = render(
      <SettingsItemCta testID={testID} title={title} ctaText="cta" />
    )

    expect(getByText(title)).toBeTruthy()
    expect(getByTestId(`${testID}/cta`)).toHaveTextContent('cta')
    expect(queryByTestId('ForwardChevron')).toBeNull()
  })

  it('renders correctly with forward chevron', () => {
    const { getByTestId, getByText } = render(
      <SettingsItemCta testID={testID} title={title} ctaText="cta" showChevron={true} />
    )

    expect(getByText(title)).toBeTruthy()
    expect(getByTestId(`${testID}/cta`)).toHaveTextContent('cta')
    expect(getByTestId('ForwardChevron')).toBeTruthy()
  })

  it('reacts on press', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <SettingsItemCta testID={testID} title={title} ctaText="cta" onPress={onPress} />
    )

    fireEvent.press(getByTestId(testID))
    expect(onPress).toHaveBeenCalled()
  })
})
