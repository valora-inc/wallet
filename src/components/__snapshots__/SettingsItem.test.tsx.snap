// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsItemSwitch renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#E6E6E6",
        "borderBottomWidth": 1,
        "marginHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          [
            {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
        >
          title
        </Text>
      </View>
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={true}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        thumbTintColor="#E6E6E6"
        trackColorForFalse="#757575"
        trackColorForTrue="#1AB775"
        trackTintColor="#1AB775"
      />
    </View>
  </View>
</View>
`;

exports[`SettingsItemTextValue renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#E6E6E6",
        "borderBottomWidth": 1,
        "marginHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          [
            {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
        >
          title
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "color": "#757575",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 24,
              "marginRight": 8,
            }
          }
          testID="testID/value"
        >
          value
        </Text>
        <svg
          fill="none"
          height={12}
          style={{}}
          testID="ForwardChevron"
          viewBox="0 0 8 16"
          width={6}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M.293.293a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1 0 1.414l-6 6a1 1 0 0 1-1.414-1.414L5.586 7 .293 1.707a1 1 0 0 1 0-1.414Z"
            fill="#757575"
            fillRule="evenodd"
            style={{}}
          />
        </svg>
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsItemTextValue renders correctly without value 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#E6E6E6",
        "borderBottomWidth": 1,
        "marginHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          [
            {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
        >
          title
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      />
    </View>
  </View>
</View>
`;
